packages/valory/skills/abstract_round_abci/models.py

```diff
@@ -67,12 +67,13 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
               validator:
                 pub_key_types: List[str]
               version: dict
+            voting_power: str
 
         :param args: positional arguments
         :param kwargs: keyword arguments
         """
         self.genesis_config = self._ensure("genesis_config", kwargs)
-        self.voting_power = self._ensure("voting_power", kwargs)
+        self.voting_power = self._ensure("voting_power", self.genesis_config)
```

Comment:
this should be named `genesis_config` then write, instead of `voting_power`, or what's the reason not to?

Reply:
This is because of the way the `voting_power` is used and the `genesis_data` have been defined here:

https://github.com/valory-xyz/open-autonomy/blob/02c0af800a4f0bb84170ddd17b5216502903d130/packages/valory/skills/registration_abci/behaviours.py#L295-L315

Therefore, I chose to use separate params for these two.

Reply:
yes i would prefer something else but let's not linger on this, we anyway need to address this in the near future again when we actually try to use this